### PR TITLE
Update DeepSleepLock

### PR DESCRIPTION
**Pull request by c1728p9**
Update the DeepSleepLock page along with moving content from this page to a new sleep manager page.

### CODE REVIEW

**Comment by c1728p9** [comment]
@sg- can you review this?

**Comment by c1728p9** [comment]
@sg- I updated this PR to talk about the constructor and destructor of RAII objects as #298 does.
CC: @deepikabhavnani

File: docs/reference/api/platform/SleepManager.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- `CAN`.
+- `SerialBase`.
+
+### Sleep manager function reference
+
+[https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_sleep.h](https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_sleep.h)
+
+### Example
+
+[![View code](https://os.mbed.com/users/c1728p9/code/SleepManager_Example_1/)](https://os.mbed.com/users/c1728p9/code/SleepManager_Example_1/file/e85412b4147e/main.cpp)
```
Comment (AnotherButler) [inline docs/reference/api/platform/SleepManager.md]: Query: Is there any way to put this in the same teams location as the other examples?

Comment (c1728p9) [reply docs/reference/api/platform/SleepManager.md]: Yep, I updated this to point to that.

**Pull request merged**